LightAPRS-hab/LightAPRS-hab.ino

```diff
@@ -106,90 +171,201 @@ void setup() {
   float pressure;
-   wdt_reset();
-  
-  if (readBatt() > BattMin) {
-  
-  if(aliveStatus){
+  float loop_start = millis();
+
+  wdt_reset();
 
+  if (readBatt() > BattMin) {
+    if(aliveStatus) {
       //send status tx on startup once (before gps fix)
-      
       #if defined(DEVMODE)
         Serial.println(F("Sending"));
       #endif
+      
       sendStatus();
+      
       #if defined(DEVMODE)
         Serial.println(F("Sent"));
       #endif
-      
+
       aliveStatus = false;
-      
-   }
-    
-    updateGpsData(1000);
-    gpsDebug();
-
-    //debug for cjr
-    //sendStatus();
-    tempC = bmp.readTemperature();
-    pressure = bmp.readPressure();
-
-    if ((gps.location.age() < 1000 || gps.location.isUpdated()) && gps.location.isValid()) {
-      if (gps.satellites.isValid() && (gps.satellites.value() > 3)) {
-      updatePosition();
-      updateTelemetry();
-      
-      //GpsOFF;
-      setGPS_PowerSaveMode();
-      //GpsFirstFix=true;
+    }
 
-      if(autoPathSizeHighAlt && gps.altitude.feet()>3000){
+    if(secsTillTx <= 0) {
+      last_tx_millis = millis();
+      secsTillTx = GPSWait;
+
+      updateGpsData(1000);
+      gpsDebug();
+
+      if ((gps.location.age() < 1000 || gps.location.isUpdated()) && 
+          gps.location.isValid()) {
+        
+        if (gps.satellites.isValid() && 
+           gps.satellites.value() > 3) {
+
+          doAltCheck();
+          updateZone(); //updates BeaconWait
+          updateComment();
+          updatePosition();
+          updateTelemetry();
+          secsTillTx = BeaconWait;
+
+          //GpsOFF;
+          setGPS_PowerSaveMode();
+          //GpsFirstFix=true;
+          current_altitude = gps.altitude.feet();
+          if(current_altitude > max_altitude) {
+            max_altitude = current_altitude;
+          }
+          if(autoPathSizeHighAlt && gps.altitude.feet() > 3000){
             //force to use high altitude settings (WIDE2-n)
             APRS_setPathSize(1);
-        } else {
+          } else {
             //use defualt settings  
             APRS_setPathSize(pathSize);
-        }
-      
-      //send status message every 60 minutes
-      if(gps.time.minute() == 30){               
-        sendStatus();       
+          }
+          
+          //send status message every 15 minutes
+          if((gps.time.minute() % 15) == 0) {               
+            sendStatus();       
+          } else {
+            sendLocation();
+          }
+
+          freeMem();
+          Serial.flush();
+        } // if time to tx
+        // sleepSeconds(BeaconWait-((millis-loop_start)/1000));
       } else {
+#if defined(DEVMODE)
+      Serial.println(F("Not enough satelites"));
+#endif
+      }
+    }
 
-         sendLocation();
+    secsTillTx -= round((millis()-loop_start)/1000);
+  } else {
+    secsToCheckBatt--;
 
-      }
+    secsToCheckBatt -= (millis()-loop_start)/1000;
+    // sleepSeconds(BattWait-((millis-loop_start)/1000));
```

Comment:
this needs to be reviewed.  we are changing secsToCheckBatt but never sleeping as a result.

Reply:
Can we talk about this?